python/ray/data/tests/test_batch_mapper.py

```diff
@@ -0,0 +1,331 @@
+import numpy as np
+import pandas as pd
+import pyarrow as pa
+
+import pytest
+from typing import Dict, Union
+from pandas.testing import assert_frame_equal
+
+import ray
+from ray.data.preprocessors import BatchMapper
+from ray.air.constants import TENSOR_COLUMN_NAME
+from ray.air.util.tensor_extensions.arrow import ArrowTensorArray
+
+
+# ===== Pandas dataset formats =====
+def ds_pandas_single_column_format():
+    in_df = pd.DataFrame({"column_1": [1, 2, 3, 4]})
+    ds = ray.data.from_pandas(in_df)
+    return ds
+
+
+def ds_pandas_multi_column_format():
+    in_df = pd.DataFrame({"column_1": [1, 2, 3, 4], "column_2": [1, -1, 1, -1]})
+    ds = ray.data.from_pandas(in_df)
+    return ds
+
+
+# ===== Arrow dataset formats =====
+def ds_arrow_single_column_format():
+    ds = ray.data.from_arrow(pa.table({"column_1": [1, 2, 3, 4]}))
+    return ds
+
+
+def ds_arrow_single_column_tensor_format():
+    ds = ray.data.from_arrow(
+        pa.table(
+            {TENSOR_COLUMN_NAME: ArrowTensorArray.from_numpy(np.array([1, 2, 3, 4]))}
+        )
+    )
+    return ds
+
+
+def ds_arrow_multi_column_format():
+    ds = ray.data.from_arrow(
+        pa.table(
+            {
+                "column_1": [1, 2, 3, 4],
+                "column_2": [1, -1, 1, -1],
+            }
+        )
+    )
+    return ds
+
+
+# ===== Numpy dataset formats =====
+def ds_numpy_single_column_tensor_format():
+    ds = ray.data.from_numpy(np.array([1, 2, 3, 4]))
+    return ds
+
+
+def ds_numpy_list_of_ndarray_tensor_format():
+    ds = ray.data.from_numpy([np.array([1, 2, 3, 4]), np.array([1, -1, 1, -1])])
+    return ds
```

Comment:
These NumPy ndarrays should have an outer batch (row) dimension and a tensor for each row/element, since that's the more realistic use case for this API.
```suggestion
    ds = ray.data.from_numpy(np.arange(12).reshape((3, 2, 2)))
    return ds


def ds_numpy_list_of_ndarray_tensor_format():
    ds = ray.data.from_numpy([np.arange(12).reshape((3, 2, 2)), np.arange(12, 24).reshape((3, 2 2))])
    return ds
```